cmd/stat.go

```diff
@@ -202,10 +209,19 @@ func statURL(ctx context.Context, targetURL, versionID string, isIncomplete, isR
 			}
 		}
 
-		_, stat, err := url2Stat(ctx, url, versionID, true, encKeyDB, time.Time{})
+		clnt, stat, err := url2Stat(ctx, url, versionID, true, encKeyDB, time.Time{})
 		if err != nil {
 			stat = content
 		}
+		// if stat is on a bucket and non-recursive mode, serve the bucket metadata
+		if clnt != nil && clnt.GetURL().Type == objectStorage && !isRecursive && stat.Type.IsDir() {
```

Comment:
I guess you can remove 'clnt.GetURL().Type == objectStorage' and 'stat.Type.IsDir()' here since you are testing for 'err == nil' here